hack/testing/rsyslog/roles/rsyslog/tasks/main.yaml

```diff
@@ -0,0 +1,175 @@
+---
+- name: Install required packages
+  yum:
+    name: '{{ item }}'
+    state: 'present'
+  with_flattened:
+    - '{{ rsyslog__viaq_prereq_packages }}'
+    - '{{ rsyslog__base_packages }}'
+    - '{{ rsyslog__viaq_packages if rsyslog__viaq|bool else [] }}'
+    - '{{ rsyslog__tls_packages if rsyslog__pki|bool else [] }}'
+    - '{{ rsyslog__packages }}'
+  when: rsyslog__enabled|bool
+
+- name: Create required system group
+  group:
+    name: '{{ rsyslog__group }}'
+    state: 'present'
+    system: True
+  when: rsyslog__unprivileged|bool and rsyslog__group != 'root'
+
+- name: Create required system user
+  user:
+    name: '{{ rsyslog__user }}'
+    group: '{{ rsyslog__group }}'
+    home: '{{ rsyslog__home }}'
+    shell: '/bin/false'
+    state: 'present'
+    createhome: False
+    system: True
+  when: rsyslog__unprivileged|bool and rsyslog__user != 'root'
+
+# Creating a work directory
+- name: Create a work directory
+  file:
+    state: directory
+    path: "{{ rsyslog__work_directory }}"
+    mode: 0700
+  changed_when: False
+
+# Creating a backup dir for rsyslog.d
+- name: Create a temp directory for rsyslog.d backup
+  command: mktemp -d /tmp/rsyslog.d-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- debug: var=mktemp.stdout
+
+- name: Set backup dir name
+  set_fact:
+    tempdir: "{{ mktemp.stdout }}"
+
+- name: Create a backup dir
+  file:
+    state: directory
+    path: "{{ tempdir }}"
+    mode: 0755
+  changed_when: False
+
+# Moving the contents of /etc/rsyslog.d to the backup dir
+- name: Moving the contents of /etc/rsyslog.d to the backup dir
+  shell: mv /etc/rsyslog.d/* "{{ tempdir }}"
+
+- name: create rsyslog viaq subdir
+  file: path="{{ rsyslog__viaq_path }}" state=directory mode=0700
+
+- name: Fix directory permissions if needed
+  file:
+    path: '/var/spool/rsyslog'
```

Comment:
I think this is a debian-ism.  We use `rsyslog__work_directory` (which is usually `/var/lib/rsyslog`) on rhel/fedora/centos.